test/unified_memory_allocation/memoryProviderAPI.cpp

```diff
@@ -77,7 +76,8 @@ INSTANTIATE_TEST_SUITE_P(
     providerInitializeTest, providerInitializeTest,
     ::testing::Values(UMA_RESULT_ERROR_OUT_OF_HOST_MEMORY,
                       UMA_RESULT_ERROR_POOL_SPECIFIC,
-                      UMA_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC,
+                      UMA_RESULT_ERROR_METADATA_MEMORY_PROVIDER_SPECIFIC,
+                      UMA_RESULT_ERROR_DATA_MEMORY_PROVIDER_SPECIFIC,
```

Comment:
I think we need a test with memory provider which always returns the error. Should it be part of this PR or we are adding tests in separate PRs?

Reply:
You mean `memory pool` that returns this error? Those two errors are supposed to be returned by `umaPoolGetLastAllocationError` (from my second PR) to indicate that the error is provider-specific.

There is some overlap between those 2 PR so I guess once one of the is merged, I will rebase (and I can add those tests) to the second one.